File: Backend/src/main/java/com/unesco/core/entities/schedule/Lesson.java
package com.unesco.core.entities.schedule;

import com.unesco.core.entities.account.Professor;

import javax.persistence.*;

@Entity
@Table(name = "un_lesson")
public class Lesson {

    @Id
    @SequenceGenerator(name = "lessonSequenceGen", sequenceName = "lessonSequenceGen", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "lessonSequenceGen")
    private long id;
    @ManyToOne
    @JoinColumn(name = "discipline_id", referencedColumnName = "id")
    private Discipline discipline;
    @ManyToOne
    @JoinColumn(name = "professor_id", referencedColumnName = "id")
    private Professor professor;
    @ManyToOne
    @JoinColumn(name = "group_id", referencedColumnName = "id")
    private Group group;

    public long getId() {
        return id;
    }
    public void setId(long id) {
        this.id = id;
    }

    public Discipline getDiscipline() {
        return discipline;
    }
    public void setDiscipline(Discipline discipline) {
        this.discipline = discipline;
    }

    public Professor getProfessor() {
        return professor;
    }
    public void setProfessor(Professor professor) {
        this.professor = professor;
    }

    public Group getGroup() {
        return group;
    }
    public void setGroup(Group group) {
        this.group = group;
    }

}

File: Frontend/Core/src/app/models/shedule/lesson.ts
import {Discipline} from "./discipline";
import {Professor} from "../account/professor";
import {Group} from "./group";
import {SemesterNumberYear} from "../semesterNumberYear.model";

export class Lesson {

    public id: number;
    public discipline: Discipline;
    public professor: Professor;
    public group: Group;
    public semesterNumberYear: SemesterNumberYear;

    constructor() {
        this.id = 0;
        this.professor = new Professor();
        this.discipline = new Discipline();
        this.group = new Group();
        this.semesterNumberYear = new SemesterNumberYear();
    }
}

export class ScheduleShowedLesson {
    public id: number;
    public discipline: Discipline;
    public professor: Professor;
    public groups: Array<Group>;
    public semesterNumberYear: SemesterNumberYear;

    constructor(lesson: Lesson) {
        this.groups = [];
        this.id = lesson.id;
        this.discipline = lesson.discipline;
        this.groups.push(lesson.group);
        this.professor = lesson.professor;
        this.semesterNumberYear = new SemesterNumberYear();
    }

}
